inject.js

```diff
@@ -334,9 +359,38 @@ function initializeWhenReady(document) {
       return;
     }
   });
+  return blacklisted;
+}
 
-  if (blacklisted) return;
+var coolDown = false;
+function refreshCoolDown() {
+  log("Begin refreshCoolDown", 5);
+  if (coolDown) {
+    clearTimeout(coolDown);
+  }
+  coolDown = setTimeout(function() {
+    coolDown = false;
+  }, 1000);
+  log("End refreshCoolDown", 5);
+}
 
+function initializeWhenReady(document) {
+  log("Begin initializeWhenReady", 5);
+  if (isBlacklisted()) {
+    return;
+  }
+  document.body.addEventListener(
+    "ratechange",
+    function(event) {
+      controller = event.target.parentElement.querySelector(".vsc-controller");
+      rateChanged(controller);
```

Comment:
I moved the original logic found in our ratechange into this function `rateChanged`. Since it is no longer being executed as part of controller's object model, I had to refactor it to use the event to find the controller rather than simply being able to use this.video. This allowed me to combine the two.

Originally, the `rateChanged` function was critical because it was manually triggered by speed adjustments via `setSpeed`, however, after combining them this is no longer a critical step.